src/resolvers-cassandra/Settings/mutations.js

```diff
@@ -231,57 +231,63 @@ function removeSite(args, res) { // eslint-disable-line no-unused-vars
 }
 
 function paramEntryToMap(paramEntry) {
-  return paramEntry.reduce((obj, item) => (obj[item.key] = item.value, obj) , {});
+  return paramEntry.reduce((obj, item) => (obj[item.key] = item.value, obj), {});
 }
 
 /**
- * @param {{input: {pipelineKey: string, pipelineLabel: string, pipelineIcon: string, streamFactory: string, params: Array<{String: String}>}}} args
+ * @param {{input: {pipelineKey: string, pipelineLabel: string, pipelineIcon: string, streamFactory: string, params: Array<{String: String}>, enabled: boolean}}} args
  * @returns {Promise}
  */
-function createStream(args, res) { // eslint-disable-line no-unused-vars
+function modifyStreams(args, res) { // eslint-disable-line no-unused-vars
   return new Promise((resolve, reject) => {
-    const pipelineKey = args && args.input && args.input.pipelineKey;
-    if (!pipelineKey || !pipelineKey.length) return reject('No pipelineKey specified.');
-    const params = paramEntryToMap(args.input.params);
+    const streams = args && args.input && args.input.streams;
+    if (!streams || !streams.length) return reject('No streams specified');
+    
+    const deletions = [];
+    const insertions = [];
+    streams.forEach(stream => {
+      deletions.push({
+        query: 'DELETE FROM fortis.streams WHERE streamid = ? AND pipelinekey = ?',
+        params: [stream.streamId, stream.pipelineKey]
+      });
 
-    cassandraConnector.executeBatchMutations([{
-      query: 'DELETE FROM fortis.streams WHERE pipelinekey = ?',
-      params: [pipelineKey]
-    }])
-    .then(() => {
-      return cassandraConnector.executeBatchMutations([{
+      let params = paramEntryToMap(stream.params);
+      insertions.push({
         query: `INSERT INTO fortis.streams (
-        streamid, 
+        streamid,
         pipelinekey,
         pipelinelabel,
         pipelineicon,
         streamfactory,
-        params
-        ) VALUES (?, ?, ?, ?, ?, ?)`,
+        params,
+        enabled
+        ) VALUES (?, ?, ?, ?, ?, ?, ?)`,
         params: [
-          uuid(), 
-          pipelineKey,
-          args.input.pipelineLabel,
-          args.input.pipelineIcon,
-          args.input.streamFactory,
-          params
+          stream.streamId, 
+          stream.pipelineKey,
+          stream.pipelineLabel,
+          stream.pipelineIcon,
+          stream.streamFactory,
+          params,
+          stream.enabled
         ]
-      }]);
+      });
+    });
+
+    cassandraConnector.executeBatchMutations(deletions)
+    .then(() => {
+      return cassandraConnector.executeBatchMutations(insertions);
```

Comment:
This `delete` + `insert` pair works. If you end up doing another commit, may I suggest changing this to an `update`?

Reply:
Done in [f3cc39d](https://github.com/CatalystCode/project-fortis-services/pull/128/commits/f3cc39de0692c5dd52475eb6bfeb002617cd2cda)